code/src/Components/MovieList.js

```diff
@@ -0,0 +1,390 @@
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(878))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'TVmovie') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10770))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10770))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'Thriller') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(53))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(53))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'War') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10752))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(10752))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		} else if (filter === 'Western') {
+			const filteredList = [...movieList]
+			if (sorting === 'released') {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(37))
+						.sort((a, b) => (a.release_date < b.release_date ? 1 : -1))
+				)
+			} else {
+				setfilteredMovieList(
+					filteredList
+						.filter((movie) => movie.genre_ids.includes(37))
+						.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+				)
+			}
+		}
+	}, [filter])
+
+	useEffect(() => {
+		if (sorting === 'popularity') {
+			const sortedMovieList = [...filteredMovieList]
+			setfilteredMovieList(
+				sortedMovieList.sort((a, b) => (a.popularity < b.popularity ? 1 : -1))
+			)
+		} else if (sorting === 'released') {
+			const sortedMovieList = [...filteredMovieList]
+			setfilteredMovieList(
+				sortedMovieList.sort((a, b) =>
+					a.release_date < b.release_date ? 1 : -1
+				)
+			)
+		}
+	}, [sorting])
+
+	return (
+		<main>
+			<div className='filter-sorting-bar'>
+				{/* ADDED options to choose to filter based on genre */}
+				<label htmlFor='filter'>
+					<select
+						id='filter'
+						value={filter}
+						onChange={(e) => setFilter(e.target.value)}
+					>
+						{' '}
+						<option selected={true} disabled='disabled'>
+							CHOOSE GENRE &#9660;
+						</option>
```

Comment:
The options could maybe be an array in which you store all the options, then map? :)